### PR TITLE
Add workspace childs list to tab panel

### PR DESCRIPTION
**Pull request by BrahimS**
Access information related to the workspace childs list on the panel tab + Add logic to Create , update and delete a workspace child

### CODE REVIEW

File: frontend/src/components/UI/Cards/ChildsTabCard.vue [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+			updateChild,
+			confirmDeleteChild,
+		} = useChildsTabData(props);
+
+		const openCard = (item) => {
+			route.push(`/workspaces/${item}`);
+		};
+
+		const openEditionModal = (item) => {
+			console.log("item: ", item);
```
Comment (Zorin95670) [inline frontend/src/components/UI/Cards/ChildsTabCard.vue]: remove this.

File: frontend/src/components/UI/TabPanels/Tabs.vue [hunk 2]
```diff
@@ -11,6 +11,8 @@
 			>
 				<q-tab v-if="tags" dense name="tags" label="Tags" icon="sell" />
 				<q-tab dense name="products" label="Products" icon="apps" />
+				<q-tab dense name="childs" label="Childs" icon="mdi-source-branch" />
+				<q-tab dense name="gitconfig" label="Git config" icon="mdi-github" />
```
Comment (Zorin95670) [inline frontend/src/components/UI/TabPanels/Tabs.vue]: This is not related to the pull request. Please, make another pull request for this.

File: frontend/src/components/UI/TabPanels/Tabs.vue [hunk 3]
```diff
@@ -157,6 +159,130 @@
...
 					</Modal>
 				</q-tab-panel>
+
+				<q-tab-panel name="childs" class="flex q-gutter-md">
+					<div
+						class="cards_wrapper"
+						v-for="child in workspaceChilds"
+						:key="child.ID"
+					>
+						<div v-if="child.Logo">
```
Comment (Zorin95670) [inline frontend/src/components/UI/TabPanels/Tabs.vue]: Do not make a if only for one attribut.
Please: 
```
:logo="child.Logo || logo" // or something else
```

File: frontend/src/components/UI/TabPanels/Tabs.vue [hunk 4]
```diff
@@ -157,6 +159,130 @@
...
+							color="white"
+							text-color="primary"
+							icon="add"
+							class="text-primary"
+							label="New child"
+							@click.prevent="openCreationProductModal(productTeam)"
+						/>
+					</div>
+					<!-- Childs Creation dialog -->
+					<Modal class="modalGlobal" v-model="isCreationProductsOpened">
```
Comment (Zorin95670) [inline frontend/src/components/UI/TabPanels/Tabs.vue:199]: Export this modal in a component

File: frontend/src/composables/TabPanels/useChildsTabData.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
+import { ref } from "vue";
+import { useStore } from "vuex";
+import { useRouter } from "vue-router";
+import { useQuasar } from "quasar";
+
+export default function useChildsTabData(props) {
```
Comment (Zorin95670) [inline frontend/src/composables/TabPanels/useChildsTabData.js]: Rename `child` to `workspace` or `subworkspace`. Because child is not relevant, do this for all child in this pull request.

File: frontend/src/composables/TabPanels/useChildsTabData.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	};
+	const addNewChild = async () => {
+		let newChild = {
+			pid: route.currentRoute.value.params.id,
+			Name: childName.value,
+			Logo: childLogo.value,
+			ShortDescription: childShortDescription.value,
+			Description: childDescription.value,
+		};
+		try {
```
Comment (Zorin95670) [inline frontend/src/composables/TabPanels/useChildsTabData.js]: don't mix try catch and promise.

File: frontend/src/composables/TabPanels/useChildsTabData.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+		let updates = {
+			ID: props.id,
+			ParentID: childParentID.value,
+			Name: childName.value,
+			Logo: childLogo.value,
+			ShortDescription: childShortDescription.value,
+			GitURL: childRepositoryURL.value,
+			Description: childDescription.value,
+		};
+		try {
```
Comment (Zorin95670) [inline frontend/src/composables/TabPanels/useChildsTabData.js]: don't mix try catch and promise.

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: frontend/src/store/Domains/actions.js [hunk 8]
```diff
@@ -19,7 +19,7 @@ export const addDomain = async ({ commit }, domain) => {
 	commit("NEW_DOMAIN", response.data);
 };
 export const updateDomain = async ({ commit }, domain) => {
-	let response = await API.put(`/domain/${domain.id}`, domain);
```
Comment (Zorin95670) [inline frontend/src/store/Domains/actions.js:22]: Export the rename of  `id` to `ID` in another commit.

File: frontend/src/components/UI/Cards/ChildsTabCard.vue (hunk 1 above)
Comment (BrahimS) [reply frontend/src/components/UI/Cards/ChildsTabCard.vue]: Ok

File: frontend/src/components/UI/TabPanels/Tabs.vue (hunk 2 above)
Comment (BrahimS) [reply frontend/src/components/UI/TabPanels/Tabs.vue]: Ok

File: frontend/src/components/UI/TabPanels/Tabs.vue (hunk 3 above)
Comment (BrahimS) [reply frontend/src/components/UI/TabPanels/Tabs.vue]: Ok, I'ill do this : :logo="child.Logo ? child.Logo : logo"

File: frontend/src/components/UI/TabPanels/Tabs.vue (hunk 4 above)
Comment (BrahimS) [reply frontend/src/components/UI/TabPanels/Tabs.vue:199]: Modal is already a component with slots and templates --->  import Modal from "components/UI/Dialogs/Modal.vue";

File: frontend/src/composables/TabPanels/useChildsTabData.js (hunk 5 above)
Comment (BrahimS) [reply frontend/src/composables/TabPanels/useChildsTabData.js]: Ok

File: frontend/src/composables/TabPanels/useChildsTabData.js (hunk 6 above)
Comment (BrahimS) [reply frontend/src/composables/TabPanels/useChildsTabData.js]: ok

File: frontend/src/composables/TabPanels/useChildsTabData.js (hunk 7 above)
Comment (BrahimS) [reply frontend/src/composables/TabPanels/useChildsTabData.js]: ok

File: frontend/src/store/Domains/actions.js (hunk 8 above)
Comment (BrahimS) [reply frontend/src/store/Domains/actions.js:22]: ok

File: frontend/src/components/UI/TabPanels/Tabs.vue (hunk 4 above)
Comment (Zorin95670) [reply frontend/src/components/UI/TabPanels/Tabs.vue:199]: Yes I know but we can have an component like `WorspaceCreationModal` with all the html/js to avoid overcharge the tabs.vue

Comment (Zorin95670) [reply frontend/src/components/UI/TabPanels/Tabs.vue:199]: And in a future, add all function of workspace in a composable if we need to reuse it :)

**Review by Zorin95670 — CHANGES_REQUESTED** [review]
Many bug on this pr:
- not blocking on empty mandatory field in subworkspace modal
- When we create subworspace on root, we see it on all workspaces instead on only the root workspace
- After creating subworking we did'nt see it in the list

**Comment by BrahimS** [comment]
In fact the bug is related to the domain store, i will create a dedicated store for sub workspaces
